src/robot/handler/protobuf.py

```diff
@@ -24,12 +27,16 @@ async def dispatch_message(self, bus_message: BusMessage) -> None:
         type_msg = bus_message.WhichOneof("message_content")
         if type_msg == "heartbeat":
             pass
+        elif type_msg == "movementEnded":
+            self.localization_controller.movement_done()
         elif type_msg == "encoderPosition":
-            if bus_message.encoderPosition.right_tick and bus_message.encoderPosition.left_tick:
-                pass
+            self.localization_controller.update_odometry_position(
+                bus_message.encoderPosition.left_tick,
+                bus_message.encoderPosition.right_tick)
```

Comment:
What do you do if you do not have all the parameters ?

Reply:
I always have all the parameters.
It's the encoder responsability to put all the necessary fields in the message. If a field is not provided, protobuf libraries usually assume that the caller want to pass the zero value in the field.

TLDR: no risk of exception here